src/vk_mem_alloc.cppm

```diff
@@ -1,70 +1,42 @@
+
+#include <vulkan/vulkan_hpp_macros.hpp>
+
+#if !VMA_USE_VULKAN_HPP_MODULE
+#include <vulkan/vulkan.hpp>
+#endif
+
+#if !VMA_USE_STD_MODULE
+#include <string>
+#include <vector>
+#endif
+
 export module vk_mem_alloc_hpp;
 
-export namespace VMA_HPP_NAMESPACE {
-  using VMA_HPP_NAMESPACE::operator|;
-  using VMA_HPP_NAMESPACE::operator&;
-  using VMA_HPP_NAMESPACE::operator^;
-  using VMA_HPP_NAMESPACE::operator~;
-  using VMA_HPP_NAMESPACE::to_string;
-  using VMA_HPP_NAMESPACE::functionsFromDispatcher;
-  using VMA_HPP_NAMESPACE::AllocatorCreateFlagBits;
-  using VMA_HPP_NAMESPACE::AllocatorCreateFlags;
-  using VMA_HPP_NAMESPACE::MemoryUsage;
-  using VMA_HPP_NAMESPACE::AllocationCreateFlagBits;
-  using VMA_HPP_NAMESPACE::AllocationCreateFlags;
-  using VMA_HPP_NAMESPACE::PoolCreateFlagBits;
-  using VMA_HPP_NAMESPACE::PoolCreateFlags;
-  using VMA_HPP_NAMESPACE::DefragmentationFlagBits;
-  using VMA_HPP_NAMESPACE::DefragmentationFlags;
-  using VMA_HPP_NAMESPACE::DefragmentationMoveOperation;
-  using VMA_HPP_NAMESPACE::VirtualBlockCreateFlagBits;
-  using VMA_HPP_NAMESPACE::VirtualBlockCreateFlags;
-  using VMA_HPP_NAMESPACE::VirtualAllocationCreateFlagBits;
-  using VMA_HPP_NAMESPACE::VirtualAllocationCreateFlags;
-  using VMA_HPP_NAMESPACE::DeviceMemoryCallbacks;
-  using VMA_HPP_NAMESPACE::VulkanFunctions;
-  using VMA_HPP_NAMESPACE::AllocatorCreateInfo;
-  using VMA_HPP_NAMESPACE::AllocatorInfo;
-  using VMA_HPP_NAMESPACE::Statistics;
-  using VMA_HPP_NAMESPACE::DetailedStatistics;
-  using VMA_HPP_NAMESPACE::TotalStatistics;
-  using VMA_HPP_NAMESPACE::Budget;
-  using VMA_HPP_NAMESPACE::AllocationCreateInfo;
-  using VMA_HPP_NAMESPACE::PoolCreateInfo;
-  using VMA_HPP_NAMESPACE::AllocationInfo;
-  using VMA_HPP_NAMESPACE::AllocationInfo2;
-  using VMA_HPP_NAMESPACE::DefragmentationInfo;
-  using VMA_HPP_NAMESPACE::DefragmentationMove;
-  using VMA_HPP_NAMESPACE::DefragmentationPassMoveInfo;
-  using VMA_HPP_NAMESPACE::DefragmentationStats;
-  using VMA_HPP_NAMESPACE::VirtualBlockCreateInfo;
-  using VMA_HPP_NAMESPACE::VirtualAllocationCreateInfo;
-  using VMA_HPP_NAMESPACE::VirtualAllocationInfo;
-  using VMA_HPP_NAMESPACE::createAllocator;
-  using VMA_HPP_NAMESPACE::createVirtualBlock;
-  using VMA_HPP_NAMESPACE::Allocator;
-  using VMA_HPP_NAMESPACE::Pool;
-  using VMA_HPP_NAMESPACE::Allocation;
-  using VMA_HPP_NAMESPACE::DefragmentationContext;
-  using VMA_HPP_NAMESPACE::VirtualAllocation;
-  using VMA_HPP_NAMESPACE::VirtualBlock;
-}
+#if VMA_USE_VULKAN_HPP_MODULE
+import vulkan_hpp;
+#endif
 
-#ifndef VULKAN_HPP_NO_SMART_HANDLE
-export namespace VMA_HPP_NAMESPACE {
-  using VMA_HPP_NAMESPACE::UniqueBuffer;
-  using VMA_HPP_NAMESPACE::UniqueImage;
-  using VMA_HPP_NAMESPACE::UniqueAllocator;
-  using VMA_HPP_NAMESPACE::UniquePool;
-  using VMA_HPP_NAMESPACE::UniqueAllocation;
-  using VMA_HPP_NAMESPACE::UniqueVirtualAllocation;
-  using VMA_HPP_NAMESPACE::UniqueVirtualBlock;
-}
+#if VMA_USE_STD_MODULE
+import std.compat;
```

Comment:
we should use VULKAN_HPP_STD_MODULE here like in vulkan_hpp module https://github.com/KhronosGroup/Vulkan-Hpp/blob/ed3cf7ac51be4c4de0b2e668da3c2f268024b2d2/vulkan/vulkan_hpp_macros.hpp#L91

Reply:
fair point, I had done it before the vulkan-hpp change. If time permits I will probably rewrite the whole PR in style that vulkan-hpp took